### PR TITLE
Arnej/global phase data from proxy

### PR DESCRIPTION
**Pull request by arnej27959**
@bjorncs please review
@bratseth FYI

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java [hunk 1]
```diff
@@ -50,4 +58,49 @@ public Model modelForRankProfile(String rankProfile) {
...
+            Supplier<FunctionEvaluator> functionEvaluatorSource = null;
+            int rerankCount = -1;
+            List<String> needInputs = null;
+
+            for (var prop : rp.fef().property()) {
+                if (prop.name().equals("vespa.globalphase.rerankcount")) {
+                    try {
+                        rerankCount = Integer.valueOf(prop.value());
+                    } catch (NumberFormatException e) {
+                        logger.warning("bad vespa.globalphase.rerankcount '" + prop.value() +
```
Comment (bjorncs) [inline container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java:84]: It's a safe assumption to fail the component (graph) construction if config is invalid.

File: container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java [hunk 2]
```diff
@@ -50,4 +58,49 @@ public Model modelForRankProfile(String rankProfile) {
...
+                    }
+                }
+                if (prop.name().equals("vespa.rank.globalphase")) {
+                    try {
+                        var model = modelForRankProfile(name);
+                        functionEvaluatorSource = () -> model.evaluatorOf("globalphase");
+                        var evaluator = functionEvaluatorSource.get();
+                        needInputs = List.copyOf(evaluator.function().arguments());
+                    } catch (IllegalArgumentException e) {
+                        logger.warning("failed setting up global-phase for " + name + " because: " + e.getMessage());
```
Comment (bjorncs) [inline container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java:95]: See comment above

File: container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java [hunk 3]
```diff
@@ -50,4 +58,49 @@ public Model modelForRankProfile(String rankProfile) {
...
         return modelForRankProfile(rankProfile).evaluatorOf(functionName);
     }
+
+    static record GlobalPhaseData(Supplier<FunctionEvaluator> functionEvaluatorSource,
+                                  int rerankCount,
+                                  List<String> needInputs) {}
+
+    private Map<String, GlobalPhaseData> profilesWithGlobalPhase = new HashMap<>();
+
+    GlobalPhaseData getGlobalPhaseData(String rankProfile) {
```
Comment (bjorncs) [inline container-search/src/main/java/com/yahoo/search/ranking/RankProfilesEvaluator.java:68]: Consider using `Optional<GlobalPhaseData>` as return type to signify that "empty" is a valid return value

File: container-search/src/main/java/com/yahoo/search/ranking/GlobalPhaseRanker.java [hunk 4]
```diff
@@ -46,9 +46,10 @@ public void process(Query query, Result result, String schema) {
             }
             return simple;
         };
-        // TODO need to get rerank-count somehow
-        int rerank = 7;
-        ResultReranker.rerankHits(result, new HitRescorer(supplier), rerank);
+        int rerankCount = data.rerankCount();
+        if (rerankCount < 0)
```
Comment (bjorncs) [inline container-search/src/main/java/com/yahoo/search/ranking/GlobalPhaseRanker.java:50]: Consider moving the default rerankCount to `config-model`

**Review by bjorncs — DISMISSED** [review]

File: container-search/src/main/java/com/yahoo/search/ranking/GlobalPhaseRanker.java (hunk 4 above)
Comment (arnej27959) [reply container-search/src/main/java/com/yahoo/search/ranking/GlobalPhaseRanker.java:52]: it's probably easier to find if it's here, I'll keep it as-is for now.

**Review by bjorncs — APPROVED** [review]

**Pull request merged**